video/out/opengl/hwdec_cuda.c

```diff
@@ -129,7 +156,7 @@ static int cuda_create(struct gl_hwdec *hw)
 
     p->hwctx = (struct mp_hwdec_ctx) {
         .type = HWDEC_CUDA,
-        .ctx = cuda_ctx,
+        .ctx = display_ctx,
         .av_device_ref = hw_device_ctx,
```

Comment:
You're right, I was confusing myself. This field value is not used, and should proably match the av_device_ref. Will change back.